.github/workflows/model-evaluation.yml

```diff
@@ -0,0 +1,159 @@
+        if: ${{ env.AZURE_CLIENT_ID != '' }}
+        run: |
+          azd auth login `
+            --client-id "$Env:AZURE_CLIENT_ID" `
+            --federated-credential-provider "github" `
+            --tenant-id "$Env:AZURE_TENANT_ID"
+        shell: pwsh
+
+      - name: Log in with Azure (Client Credentials)
+        if: ${{ env.AZURE_CREDENTIALS != '' }}
+        run: |
+          $info = $Env:AZURE_CREDENTIALS | ConvertFrom-Json -AsHashtable;
+          Write-Host "::add-mask::$($info.clientSecret)"
+
+          azd auth login `
+            --client-id "$($info.clientId)" `
+            --client-secret "$($info.clientSecret)" `
+            --tenant-id "$($info.tenantId)"
+        shell: pwsh
+        env:
+          AZURE_CREDENTIALS: ${{ secrets.AZURE_CREDENTIALS }}
+
+      #-------------------- Deploy Application --------------------
+
+      - name: Provision and deploy application
+        if: ${{ !inputs.skip-deploy }}
+        run: |
+          azd up --no-prompt
+
+      - name: Refresh azd environment
+        run: |
+          azd env refresh --no-prompt
+
+      #---------------------- Run Evaluation ----------------------
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r evaluation/requirements.txt
+
+      - name: Create evaluation environment
+        run: |
+          ./scripts/create_eval_dotenv.sh
+
+      - name: Generate Q&As
+        run: |
+          python -m evaluation generate --numquestions ${{ inputs.eval-num-questions }}
+
+      - name: Run AI RAG Chat evaluation
+        run: |
+          python -m evaluation evaluate --numquestions ${{ inputs.eval-num-questions }}
+
+          # Store evaluation results path
+          results_path=$(ls -d evaluation/results/gpt_evaluation/experiment-* | tail -n 1)
+          echo "EVALUATION_RESULTS=$results_path" >> $GITHUB_ENV
+
+      - name: Run Red Teaming evaluation
+        run: |
+          python -m evaluation red-teaming
+
+          # Store red teaming results path
+          results_path=$(ls -d evaluation/results/red_teaming/experiment-* | tail -n 1)
+          echo "RED_TEAMING_RESULTS=$results_path" >> $GITHUB_ENV
+
+      - name: Dump results
+        uses: actions/upload-artifact@v4
+        with:
+          name: evaluation-results
+          path: |
+            ${{ env.EVALUATION_RESULTS }}/summary.json
+            ${{ env.EVALUATION_RESULTS }}/eval_results.jsonl
+            ${{ env.EVALUATION_RESULTS }}/config.json
+            ${{ env.EVALUATION_RESULTS }}/eval.png
+            ${{ env.EVALUATION_RESULTS }}/mean_score.png
+            ${{ env.EVALUATION_RESULTS }}/passing_rate.png
+            ${{ env.RED_TEAMING_RESULTS }}/scores.json
+
+      #------------------------- Cleanup --------------------------
+
+      - name: Purge deployment
+        if: ${{ always() && inputs.purge }}
+        run: azd down --force --purge
```

Comment:
I think I forgot to fix this, this `azd down --force --purge` isn't going to work properly, because `azd down` doesn't accept these 2 arguments at the same time, which will still require manual operation. I went to the doc before and I think `azd down --force` should be the right thing to put here, but I haven't tested that yet. Could you please verify that?

Reply:
I've been running it this way both locally and in the CI, seems to be working!